docs/architecture/grpc-for-wcf-developers/protobuf-data-types.md

```diff
@@ -1,11 +1,158 @@
 ---
 title: Protobuf scalar data types - gRPC for WCF Developers
-description: TO BE WRITTEN
+description: The basic and well-known data types supported by Protobuf and gRPC in .NET Core
 author: markrendle
-ms.date: 09/02/2019
+ms.date: 09/09/2019
 ---
 
 # Protobuf scalar data types
 
+Protobuf supports a range of native scalar value types. This table lists them all with their equivalent C# type.
+
+| Protobuf Type | C# Type    | Notes |
+| ------------- | ---------- | ----- |
+| double        | double     |       |
+| float         | float      |       |
+| int32         | int        |       |
+| int64         | long       |       |
+| uint32        | uint       |       |
+| uint64        | ulong      |       |
+| sint32        | int        | 1     |
+| sint64        | long       | 1     |
+| fixed32       | uint       | 2     |
+| fixed64       | ulong      | 2     |
+| sfixed32      | int        | 2     |
+| sfixed64      | long       | 2     |
+| bool          | bool       |       |
+| string        | string     | 3     |
+| bytes         | ByteString | 4     |
+
+## Notes
+
+1. The standard encoding for `int32` and `int64` is inefficient when working with signed values. If your field is likely to contain negative numbers, use `sint32` or `sint64` instead. Both types map to the C# `int` and `long` types respectively.
+2. The `fixed` fields always use the same number of bytes regardless of the value. This makes serialization/deserialization faster for larger values.
+3. Protobuf strings are UTF-8 (or 7-bit ASCII) encoded, and the encoded length cannot be greater than 2<sup>32</sup>.
+4. The Protobuf runtime provides a `ByteString` type that maps easily to and from C# `byte[]` arrays.
+
+## Other .NET primitive types
+
+### Dates and times
+
+The native scalar types do not provide for date and time values, equivalent to C#'s `DateTimeOffset`, `DateTime` and `TimeSpan`. These types can be specified using some of Google's "Well Known Types" extensions, which provide code generation and runtime support for more complex field types across the supported platforms. To use them, you need to import them in your `.proto` file, like this:
```

Comment:
I think another table here would be useful that maps DateTimeOffset, DateTime and TimeSpan to duration and timestamp.

Reply:
Done.